### PR TITLE
[FEAT] Deploy workflow functionality added, Built workflow steps UI 

### PR DESCRIPTION
**Pull request by yash-25log**
image
#9 #21

### CODE REVIEW

File: client/my-app/src/app/workflow/page.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    repo?: string;
+    accessToken?: string;
+}
+
+export default function WorkflowSteps({
+    owner = "",
+    repo = "",
+    accessToken = "" 
+}: WorkflowStepsProps) {
+
```
Comment (sahilrohera10) [inline client/my-app/src/app/workflow/page.tsx]: dont need to take these things from the props, rather take these from the local storage

File: client/my-app/src/app/workflow/page.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    const [repoName, setRepoName] = useState('')
+    const baseUrl = process.env.NEXT_PUBLIC_BASE_URL
+
+    useEffect(() =>{
+        const urlParams = new URLSearchParams(window.location.search)
+        const prNameFromUrl = urlParams.get('repoName')
+        if(prNameFromUrl) {
+            setRepoName(prNameFromUrl)
+        }
+    }, [])
```
Comment (sahilrohera10) [inline client/my-app/src/app/workflow/page.tsx]: change this to repoName .. also take this repoName searching out of the useEffect and write repoNameFromUrl in dev dependency

File: client/my-app/src/app/workflow/page.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+            <span className="text-l text-zinc-200">Add Email ID</span>
+          </div>
+          {showEmailInput && (
+              <Input
+                type="email"
+                placeholder="Enter your email"
+                value={email}
+                onChange={(e) => setEmail(e.target.value)}
+                className="mt-4 bg-zinc-900 border-zinc-700 text-zinc-200"
```
Comment (sahilrohera10) [inline client/my-app/src/app/workflow/page.tsx]: create a handleEmail function and in that function change the particular state

**Review by sahilrohera10 — CHANGES_REQUESTED** [review]

File: client/my-app/src/app/workflow/page.tsx (hunk 1 above)
Comment (yash-25log) [reply client/my-app/src/app/workflow/page.tsx]: done

File: client/my-app/src/app/workflow/page.tsx (hunk 2 above)
Comment (yash-25log) [reply client/my-app/src/app/workflow/page.tsx]: got it

File: client/my-app/src/app/workflow/page.tsx (hunk 3 above)
Comment (yash-25log) [reply client/my-app/src/app/workflow/page.tsx]: updated

File: client/my-app/src/app/workflow/page.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    const [showEmailInput, setShowEmailInput] = useState(false)
+    const [email, setEmail] = useState('')
+    const [isDeploying, setIsDeploying] = useState(false)
+    const [repoName, setRepoName] = useState('')
+    const baseUrl = process.env.NEXT_PUBLIC_BASE_URL
+
+    const owner = localStorage.getItem('owner') || ''
+    const repo = localStorage.getItem('repo') || ''
+    const accessToken = localStorage.getItem('accessToken') || ''
```
Comment (sahilrohera10) [inline client/my-app/src/app/workflow/page.tsx]: remove this repo

File: client/my-app/src/app/workflow/page.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            <span className="text-l text-zinc-200">Add Email ID</span>
+          </div>
+          {showEmailInput && (
+              <Input
+                type="email"
+                placeholder="Enter your email"
+                value={email}
+                onChange={handleEmail}
+                className="mt-4 bg-zinc-900 border-zinc-700 text-zinc-200"
+              />
```
Comment (sahilrohera10) [inline client/my-app/src/app/workflow/page.tsx]: fix this .. not passing event to the function and call handleEmail in anonymous function

**Review by sahilrohera10 — CHANGES_REQUESTED** [review]

File: client/my-app/src/app/workflow/page.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            email
+        }
+        try {
+            const response = await fetch(`${baseUrl}/api/v1-2024/github/register/webhook`,{
+                method: 'POST', 
+                headers:{
+                    'Content-Type': 'application/json',
+                },
+                body:JSON.stringify({payload})
+            })
```
Comment (sahilrohera10) [inline client/my-app/src/app/workflow/page.tsx]: remove this curly braces

**Review by sahilrohera10 — CHANGES_REQUESTED** [review]

File: client/my-app/src/app/workflow/page.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    }, [repoNameFromUrl])
+
+    const handleDeploy = async() => {
+        setIsDeploying(true)
+        const payload = {
+            owner,
+            repo: repoName,
+            access_token: accessToken,
+            email
+        }
```
Comment (sahilrohera10) [inline client/my-app/src/app/workflow/page.tsx]: send email as additional email and also send repo_url and user_id as well

**Review by sahilrohera10 — CHANGES_REQUESTED** [review]

**Review by sahilrohera10 — APPROVED** [review]

**Pull request merged**